### ISSUE TITLE
Uploaded fin, but now what?

### ISSUE THREAD

**Issue by lordratner** [andyboeh/esphome-sense-u#1]
Hey there,

I was able to get the ESP32 working with the yaml config, but now how do I get the device to pair?

I see all the available BT devices, and I can see the Sense-U, but it doesn't seem to do anything after that:

`
[19:03:53][I][app:102]: ESPHome version 2022.11.1 compiled on Nov 20 2022, 19:01:58
[19:03:53][C][wifi:504]: WiFi:
[19:03:53][C][wifi:362]:   Local MAC: CC:50:E3:AF:1A:E4
[19:03:53][C][wifi:363]:   SSID: [redacted]
[19:03:53][C][wifi:364]:   IP Address: 192.168.20.70
[19:03:53][C][wifi:366]:   BSSID: [redacted]
[19:03:53][C][wifi:367]:   Hostname: 'sense-u'
[19:03:53][C][wifi:369]:   Signal strength: -83 dB ▂▄▆█
[19:03:53][C][wifi:373]:   Channel: 4
[19:03:53][C][logger:293]: Logger:
[19:03:53][C][logger:294]:   Level: DEBUG
[19:03:53][C][logger:295]:   Log Baud Rate: 115200
[19:03:53][C][logger:296]:   Hardware UART: UART0
[19:03:53][C][uptime.sensor:031]: Uptime Sensor 'Sense-U Baby Tracker Uptime'
[19:03:53][C][uptime.sensor:031]:   Device Class: 'duration'
[19:03:53][C][uptime.sensor:031]:   State Class: 'total_increasing'
[19:03:53][C][uptime.sensor:031]:   Unit of Measurement: 's'
[19:03:53][C][uptime.sensor:031]:   Accuracy Decimals: 0
[19:03:53][C][senseu:013]: SenseU
[19:03:53][C][senseu:014]:    Unit of Measurement: '/min'
[19:03:53][C][senseu:014]:    Accuracy Decimals: 0
[19:03:53][C][senseu:015]:  Temperature 'Temperature'
[19:03:53][C][senseu:015]:    State Class: ''
[19:03:53][C][senseu:015]:    Accuracy Decimals: 1
[19:03:53][C][esp32_ble_tracker:793]: BLE Tracker:
[19:03:54][C][esp32_ble_tracker:794]:   Scan Duration: 300 s
[19:03:54][C][esp32_ble_tracker:795]:   Scan Interval: 320.0 ms
[19:03:54][C][esp32_ble_tracker:796]:   Scan Window: 30.0 ms
[19:03:54][C][esp32_ble_tracker:797]:   Scan Type: ACTIVE
[19:03:54][C][esp32_ble_tracker:798]:   Continuous Scanning: True
[19:03:54][C][api:138]: API Server:
[19:03:54][C][api:139]:   Address: 192.168.20.70:6053
[19:03:54][C][api:143]:   Using noise encryption: NO
[19:03:54][C][wifi_signal.sensor:009]: WiFi Signal 'Sense-U Baby Tracker Wifi Signal'
[19:03:54][C][wifi_signal.sensor:009]:   Device Class: 'signal_strength'
[19:03:54][D][api:102]: Accepted ::FFFF:C0A8:1432
[19:03:54][D][api.connection:917]: Home Assistant 2022.11.2 (::FFFF:C0A8:1432): Connected successfully
[19:03:54][D][time:045]: Synchronized time: 2022-11-20 19:03:54
[19:03:58][D][esp32_ble_tracker:809]: Found device 6A:78:91:FB:85:2F RSSI=-93
[19:03:58][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:03:59][D][esp32_ble_tracker:809]: Found device 43:DA:B6:96:FE:40 RSSI=-86
[19:03:59][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:04:05][D][esp32_ble_tracker:809]: Found device FA:FE:7C:23:6B:07 RSSI=-81
[19:04:05][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:04:05][D][esp32_ble_tracker:832]:   Name: 'Sense-U Baby'
[19:04:05][D][esp32_ble_tracker:809]: Found device 70:30:30:38:6D:FB RSSI=-65
[19:04:05][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:04:08][D][esp32_ble_tracker:809]: Found device 63:A6:04:A7:09:43 RSSI=-94
[19:04:08][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:04:10][D][esp32_ble_tracker:809]: Found device 79:E3:6E:89:44:44 RSSI=-90
[19:04:10][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:04:13][D][esp32_ble_tracker:809]: Found device 76:76:76:19:61:87 RSSI=-98
[19:04:13][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:04:18][D][esp32_ble_tracker:809]: Found device D7:08:8D:65:4A:9F RSSI=-97
[19:04:18][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:04:23][D][esp32_ble_tracker:809]: Found device FF:20:DB:7B:6F:9D RSSI=-81
[19:04:23][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:04:31][D][sensor:127]: 'Sense-U Baby Tracker Wifi Signal': Sending state -78.00000 dBm with 0 decimals of accuracy
[19:04:41][D][sensor:127]: 'Sense-U Baby Tracker Uptime': Sending state 56.41100 s with 0 decimals of accuracy
[19:05:03][D][esp32_ble_tracker:809]: Found device 4A:7F:85:A4:06:EE RSSI=-88
[19:05:03][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:05:29][D][esp32_ble_tracker:809]: Found device 45:C6:27:A5:32:87 RSSI=-85
[19:05:29][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:05:31][D][sensor:127]: 'Sense-U Baby Tracker Wifi Signal': Sending state -81.00000 dBm with 0 decimals of accuracy
[19:05:34][D][esp32_ble_tracker:809]: Found device 42:D5:C8:60:8A:CA RSSI=-82
[19:05:34][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:05:36][D][esp32_ble_tracker:809]: Found device 4F:34:A9:B1:F7:7B RSSI=-88
[19:05:36][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:05:41][D][sensor:127]: 'Sense-U Baby Tracker Uptime': Sending state 116.41100 s with 0 decimals of accuracy
[19:06:31][D][sensor:127]: 'Sense-U Baby Tracker Wifi Signal': Sending state -79.00000 dBm with 0 decimals of accuracy
[19:06:41][D][sensor:127]: 'Sense-U Baby Tracker Uptime': Sending state 176.41100 s with 0 decimals of accuracy
[19:06:46][D][esp32_ble_tracker:809]: Found device 3F:BC:CD:27:E3:06 RSSI=-93
[19:06:46][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:07:19][D][esp32_ble_tracker:809]: Found device 53:6E:70:18:CF:89 RSSI=-94
[19:07:19][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:07:31][D][sensor:127]: 'Sense-U Baby Tracker Wifi Signal': Sending state -83.00000 dBm with 0 decimals of accuracy
[19:07:41][D][esp32_ble_tracker:809]: Found device FC:89:F0:0B:3E:45 RSSI=-88
[19:07:41][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:07:41][D][sensor:127]: 'Sense-U Baby Tracker Uptime': Sending state 236.41200 s with 0 decimals of accuracy
[19:07:51][D][esp32_ble_tracker:809]: Found device 77:0A:34:8E:3B:71 RSSI=-97
[19:07:51][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:07:52][D][esp32_ble_tracker:809]: Found device 78:61:82:D7:79:A8 RSSI=-93
[19:07:52][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:07:52][D][esp32_ble_tracker:834]:   TX Power: 2
[19:08:31][D][sensor:127]: 'Sense-U Baby Tracker Wifi Signal': Sending state -80.00000 dBm with 0 decimals of accuracy
[19:08:41][D][sensor:127]: 'Sense-U Baby Tracker Uptime': Sending state 296.41101 s with 0 decimals of accuracy
[19:08:45][I][ota:113]: Boot seems successful, resetting boot loop counter.
[19:08:45][D][esp32.preferences:113]: Saving 1 preferences to flash...
[19:08:45][D][esp32.preferences:142]: Saving 1 preferences to flash: 0 cached, 1 written, 0 failed
[19:08:46][D][esp32_ble_tracker:264]: Starting scan...
[19:08:46][D][esp32_ble_tracker:809]: Found device 42:D5:C8:60:8A:CA RSSI=-79
[19:08:46][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:08:47][D][esp32_ble_tracker:809]: Found device 70:30:30:38:6D:FB RSSI=-70
[19:08:47][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:08:49][D][esp32_ble_tracker:809]: Found device 4A:C0:6D:0B:DD:E1 RSSI=-61
[19:08:49][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:08:49][D][esp32_ble_tracker:809]: Found device FA:FE:7C:23:6B:07 RSSI=-74
[19:08:49][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:08:49][D][esp32_ble_tracker:832]:   Name: 'Sense-U Baby'
[19:08:50][D][esp32_ble_tracker:809]: Found device 6F:77:9B:B4:51:47 RSSI=-79
[19:08:50][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:08:50][D][esp32_ble_tracker:809]: Found device 24:FC:E5:66:34:F9 RSSI=-88
[19:08:50][D][esp32_ble_tracker:830]:   Address Type: PUBLIC
[19:08:51][D][esp32_ble_tracker:809]: Found device 4F:34:A9:B1:F7:7B RSSI=-89
[19:08:52][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:08:53][D][esp32_ble_tracker:809]: Found device 77:0A:34:8E:3B:71 RSSI=-92
[19:08:53][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:09:00][D][esp32_ble_tracker:809]: Found device 4A:7F:85:A4:06:EE RSSI=-85
[19:09:00][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:09:02][D][esp32_ble_tracker:809]: Found device 79:E3:6E:89:44:44 RSSI=-87
[19:09:02][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:09:08][D][esp32_ble_tracker:809]: Found device 5E:C0:C4:11:25:A6 RSSI=-67
[19:09:08][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:09:13][D][esp32_ble_tracker:809]: Found device FC:89:F0:0B:3E:45 RSSI=-95
[19:09:13][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:09:26][D][esp32_ble_tracker:809]: Found device 6B:96:40:80:17:03 RSSI=-90
[19:09:26][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:09:26][D][esp32_ble_tracker:834]:   TX Power: 2
[19:09:27][D][esp32_ble_tracker:809]: Found device 7E:FE:FE:9E:5F:BB RSSI=-96
[19:09:27][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:09:31][D][sensor:127]: 'Sense-U Baby Tracker Wifi Signal': Sending state -78.00000 dBm with 0 decimals of accuracy
[19:09:41][D][sensor:127]: 'Sense-U Baby Tracker Uptime': Sending state 356.41101 s with 0 decimals of accuracy
[19:10:04][D][esp32_ble_tracker:809]: Found device 5C:C8:A2:47:CD:1E RSSI=-78
[19:10:04][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:10:06][D][esp32_ble_tracker:809]: Found device 70:F1:67:CF:0A:32 RSSI=-97
[19:10:06][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:10:06][D][esp32_ble_tracker:834]:   TX Power: 2
[19:10:11][D][esp32_ble_tracker:809]: Found device DD:3D:29:8A:D1:60 RSSI=-97
[19:10:11][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:10:11][D][esp32_ble_tracker:832]:   Name: 'N3UZM'
[19:10:17][D][esp32_ble_tracker:809]: Found device 48:C2:A7:4C:60:AE RSSI=-77
[19:10:17][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:10:31][D][sensor:127]: 'Sense-U Baby Tracker Wifi Signal': Sending state -77.00000 dBm with 0 decimals of accuracy
[19:10:36][D][esp32_ble_tracker:809]: Found device 5C:C8:C2:C2:81:97 RSSI=-76
[19:10:36][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:10:41][D][sensor:127]: 'Sense-U Baby Tracker Uptime': Sending state 416.41199 s with 0 decimals of accuracy
[19:11:00][D][esp32_ble_tracker:809]: Found device 50:0E:84:70:C9:E2 RSSI=-64
[19:11:00][D][esp32_ble_tracker:830]:   Address Type: RANDOM
[19:11:03][D][esp32_ble_tracker:809]: Found device FF:20:DB:7B:6F:9D RSSI=-82
[19:11:03][D][esp32_ble_tracker:830]:   Address Type: RANDOM
`

Not sure how I get it to initialize?

Also, is the "baby_code" the MAC address of the Sense-U?

**Comment by lordratner** [comment]
Ok, figured out that the Sense-U MAC goes in the "ble_client" mac_address. On to the next issue

**Comment by mattmcardle** [comment]
Where can I find the Sense-u MAC address?

**Comment by andyboeh** [comment]
@mattmcardle Please do not post unrelated questions to closed issues!
That said: As with any other ESPHome BLE device, in the ESPHome log. Or use any BLE scanner app for your phone.

**Comment by mattmcardle** [comment]
Apologies, I felt my query was related to the original issue, although I accept I should have raised a new issue.

I am new to ESPHome and BLE development in general. Before posting this question I attempted to use an app called BLE Scanner on iOS to get the MAC address of the sense-u, but it seemed to provide me with every other possible characteristic about the device apart from the MAC Address! Further research indicated that iOS restricts access to this information for security purposes :roll_eyes: 

In the ESPHome log the only valid MAC address was that of the WiFi network I was connecting to.

Eventually I realised I could find the sense-u MAC address from my Ubuntu desktop by simply connecting to it and then clicking on the device in the list of bluetooth devices!

Anyway, many thanks for your work on this @andyboeh, after I overcame the MAC address issue the process of connecting was painless.
